src/main/java/ui/components/pickers/LabelPickerDialog.java

```diff
@@ -3,47 +3,171 @@
 import backend.resource.TurboIssue;
 import backend.resource.TurboLabel;
 import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
 import javafx.geometry.Insets;
 import javafx.scene.control.*;
-import javafx.scene.input.KeyCode;
 import javafx.scene.layout.FlowPane;
 import javafx.scene.layout.VBox;
 import javafx.stage.Modality;
 import javafx.stage.Stage;
 
+import java.io.IOException;
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
+import com.sun.javafx.tk.FontLoader;
+import com.sun.javafx.tk.Toolkit;
+
+import ui.UI;
+
+/**
+ * A Presenter class to synchronize model and view  after each user action
+ * @author Jin
+ *
+ */
 public class LabelPickerDialog extends Dialog<List<String>> {
 
-    private static final int VBOX_SPACING = 105; // seems like some magic number
+    private static final int VBOX_SPACING = 105;
     private static final int ELEMENT_MAX_WIDTH = 400;
+    private static final String EXCLUSIVE_DELIMITER = ".";
+    private static final String NONEXCLUSIVE_DELIMITER = "-";
```

Comment:
These are already in TurboLabel. Actually, why are all the TurboLabel methods duplicated here?